### PR TITLE
packager: fetch glibc corresponding to the host architecture on debian multiarch

### PR DESCRIPTION
**Pull request by nlewycky**
*Description of changes:*
This is a Debian version of PR #120. Many Debian machines have both amd64 and i386 (64 and 32 bit x86-64) installed. When "dpkg-query --listfiles libc6" is run, dpkg doesn't know which package it's referring to:
```
dpkg-query: error: --listfiles needs a valid package name but 'libc6' is not: ambiguous package name 'libc6' with more than one installed instance

Use --help for help about querying packages.
```

The main page for more information on MultiArch is https://wiki.debian.org/Multiarch .

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packaging/packager [hunk 1]
```diff
@@ -66,14 +65,16 @@ function package_libc_via_pacman {
 
 function package_libc_via_dpkg() {
     if type dpkg-query > /dev/null 2>&1; then
-        if [[ $(dpkg-query --listfiles libc6 | wc -l) -gt 0 ]]; then
-            dpkg-query --listfiles libc6 | sed -E '/\.so$|\.so\.[0-9]+$/!d'
+       architecture=$(dpkg --print-architecture)
```
Comment (specious) [inline packaging/packager:68]: I just [added](https://github.com/awslabs/aws-lambda-cpp/pull/131) Arch Linux support in the packager the other day, and I switched to using `uname -m` for checking the system architecture for rpm-based distros.

Comment (nlewycky) [reply packaging/packager:68]: Doesn't work on Debian though:

```
$ uname -m
x86_64
$ dpkg --print-architecture
amd64
```

Using `x86_64` as the architecture does not work with Debian.

Comment (specious) [reply packaging/packager:81]: Oh, interesting.

And I just took a look with a fedora image from docker hub, and it turns out `arch` comes with `coreutils` on some distros:
```
$ arch --help
Usage: arch [OPTION]...
Print machine architecture.

      --help     display this help and exit
      --version  output version information and exit

GNU coreutils online help: <https://www.gnu.org/software/coreutils/>
Report any translation bugs to <https://translationproject.org/team/>
Full documentation <https://www.gnu.org/software/coreutils/arch>
or available locally via: info '(coreutils) arch invocation'
```
And the [docs](https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html#arch-invocation) say:
> `arch` prints the machine hardware name, and is equivalent to `uname -m`

Comment (specious) [reply packaging/packager:81]: Just want to give you a heads up, I just opened PR #136, which has some simplifications in the same area as this PR.

Comment (specious) [reply packaging/packager:81]: I cherry-picked your changes from here into #136: d63870c

**Review by bmoffatt — APPROVED** [review]
LGTM

**Pull request merged**